### PR TITLE
rbd-mirror: do not propagate deletions when pool unavailable

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/16229
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/tools/rbd_mirror/PoolWatcher.cc [hunk 1]
```diff
@@ -72,7 +74,7 @@ void PoolWatcher::refresh_images(bool reschedule)
   // about new/removed mirrored images
 }
 
-void PoolWatcher::refresh(ImageIds *image_ids) {
+int PoolWatcher::refresh(ImageIds *image_ids) {
```
Comment (trociny) [inline src/tools/rbd_mirror/PoolWatcher.cc]: @dillaman Just wander, below we are ignoring list_images_v2 failure. May be it is safer just to return error there too?

Comment (dillaman) [inline src/tools/rbd_mirror/PoolWatcher.cc]: Good catch

**Comment by dillaman** [comment]
@trociny updated

**Comment by trociny** [comment]
lgtm

**Pull request merged**